Remove call to std::set_new_handler

pierre-dejoue opened the pull request:
I'm not sure of the intention behind this call to std::set_new_handler, but the new-handler is global to the program, therefore modifying it from a library seems unwarranted.

pierre-dejoue committed:
Remove call to std::set_new_handler

m-reuter commented:
Thanks. yes, globally changing new handler would be unwanted. However, I think the intention is to call the ARPACK exception for memory overflow later? We could reset the handler to its original value afterwards? Or maybe it is OK, just to remove it (like you did)?
Is this the only place where the hander is set?

Also, I fixed the travis build (they had updated their macs and I fixed some install scripts). Can you merge those updates and update this pull request again to make sure tests work.
Thanks

pierre-dejoue commented:
Hi, and sorry for the delayed answer!

I rebased the patch on top of your latest commits. Travis checks pass this time!

To answer your questions, this is the only place where std::set_new_handler was called. But this is not the only place where the code throws an ArpackError(ArpackError::MEMORY_OVERFLOW, "..."). However if we look at the code there, the ArpackError::MEMORY_OVERFLOW is not the result of a catched std::bad_alloc, but rather a consequence of detecting that some pre-allocated memory will not fit some new data.

I'd say we have two choices:
1) Remove the call to set_new_handler entirely there (the present state of my commit). A std::bad_alloc will then be thrown to the library's user in the event of a failed mem allocation in arpackpp.
2) try ... catch a std::bad_alloc in the method where I removed set_new_handler (ARrcStdEig::DefineParameters) and throw ArpackError(ArpackError::MEMORY_OVERFLOW) instead.

Option 2) is closer to the original behavior, but I'm not sure it is the desired one. It looks in any case better than to deal with the global new_handler (as you say, we would have to store and restore it).

Let me know your opinion, I can update the PR accordingly.

Thanks!
